nautobot/core/tests/integration/test_swagger.py

```diff
@@ -0,0 +1,56 @@
+import time
+
+from django.urls import reverse
+
+from nautobot.utilities.testing.integration import SeleniumTestCase
+
+
+class SwaggerUITestCase(SeleniumTestCase):
+    """Integration tests for the Swagger UI."""
+
+    def setUp(self):
+        super().setUp()
+        self.user.is_superuser = True
+        self.user.save()
+        self.login(self.user.username, self.password)
+
+    def tearDown(self):
+        self.logout()
+        super().tearDown()
+
+    def test_endpoint_render(self):
+        """Check that the dcim.site API endpoints are rendered correctly."""
+        self.browser.visit(self.live_server_url + reverse("api_docs"))
+        # Wait for Swagger UI to load
+        load_time = 0
+        while self.browser.find_by_xpath("//div[@class[contains(., 'loading')]]"):
+            print("Waiting for schema to load...")
+            time.sleep(1)
+            load_time += 1
+            if load_time > 20:
+                self.fail("Schema took more than 20 seconds to load")
```

Comment:
I tried this but Glenn's is _slightly_ faster because it can optionally early. What he's checking for is something to disappear (hence the while found).

Profiled it locally and Glenn's exited earlier several times.

Reply:
Although now that I see how that's implemented under the hood in Splinter, I don't know why that was the case: https://github.com/cobrateam/splinter/blob/master/splinter/driver/webdriver/__init__.py#L372-L379

Reply:
Wasn't aware of that option, thanks for the pointer.

Reply:
Would be nice if it were documented at https://splinter.readthedocs.io/en/latest/finding.html ...